### PR TITLE
Add script to render screenshots of dumb components

### PR DESCRIPTION
**Pull request by chromakode**
:eyeglasses: @keybase/react-hackers 

re: DESKTOP-844

### CODE REVIEW

File: shared/more/dumb-map.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import CommonMap from '../common-adapters/dumb.desktop'
```
Comment (chromakode) [inline shared/more/dumb-map.js]: Perhaps I should rename this file (and its associated variables) to `dumb-component-map.js`

File: shared/test/render-dumb-sheet.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  )
+
+  const appEl = document.getElementById('app')
+  ReactDOM.render(displayTree, appEl, () => {
+    // Unfortunately some resources like fonts lazy load after they're
+    // rendered.  We need to give the renderer time to load.  After trying
+    // process.nextTick, requestAnimationFrame, etc., simply putting in a time
+    // delay worked the best.
+    setTimeout(() => {
+      const renderedEl = document.getElementById('rendered')
```
Comment (chromakode) [inline shared/test/render-dumb-sheet.js]: I was initially using `appEl`.firstChild but the flow type for that is `Node` -- couldn't figure out how to cast to `HTMLElement`.

File: shared/more/dumb-map.js (hunk 1 above)
Comment (MarcoPolo) [inline shared/more/dumb-map.js]: 👍

File: desktop/test/render-visdiff.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    const filenameParts = [msg.key, msg.mockKey].map(s => s.replace(' ', '_').replace(/[^\w]*/g, ''))
+    const filename = filenameParts.join('-') + '.png'
+    fs.writeFileSync(path.join(outputDir, filename), img.toPng())
+    console.log('wrote', filename)
+    renderNext()
+  })
+})
+
+app.on('ready', () => {
+  win = new BrowserWindow({show: false, width: 1000, height: 1000})
```
Comment (MarcoPolo) [inline desktop/test/render-visdiff.js]: 🐑 would be a little nicer if `win` was set once as `const` and passed around, instead of just using one instance of `win`

**Comment by MarcoPolo** [comment]
lgtm

File: desktop/test/render-visdiff.js (hunk 3 above)
Comment (chromakode) [inline desktop/test/render-visdiff.js]: Yeah I dithered back and forth on that. Will take another look.

**Comment by chromakode** [comment]
@MarcoPolo Just realized I didn't update the native dumb sheet. Should I do this here, or punt on it til later?

**Comment by MarcoPolo** [comment]
I think you should make a separate ticket for this and merge this in! Smaller PR's are always bettter :)

**Comment by chromakode** [comment]
This approach is electron-specific. Might be possible to rig a similar approach with emulators in the future, or find some app-centric CI tool that can do this.

**Comment by chromakode** [comment]
@chrisnojima However I did consolidate the dumb component maps for this pull request, and should look into giving the native dumb sheet a similar treatment.

**Comment by chrisnojima** [comment]
Sounds good

File: desktop/test/render-visdiff.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    if (!toRender.length) {
+      app.quit()
+      return
+    }
+    win.webContents.send('display', toRender.pop())
+  }
+
+  ipcMain.on('display-done', (ev, msg) => {
+    win.capturePage(msg.rect, img => {
+      const filenameParts = [msg.key, msg.mockKey].map(s => s.replace(' ', '_').replace(/[^\w]*/g, ''))
```
Comment (chrisnojima) [inline desktop/test/render-visdiff.js]: maybe just ```_.words(s)```?

File: desktop/webpack.config.base.js [hunk 5]
```diff
@@ -16,8 +16,8 @@ module.exports = {
       exclude: /(node_modules|\/dist\/)/,
       query: {
         cacheDirectory: true,
-        plugins: ['transform-runtime'],
-        presets: ['es2015', 'stage-1', 'react']
+        plugins: ['babel-plugin-transform-runtime'].map(require.resolve),
```
Comment (chrisnojima) [inline desktop/webpack.config.base.js]: why is this necessary?

File: shared/test/render-dumb-sheet.js (hunk 2 above)
Comment (chrisnojima) [inline shared/test/render-dumb-sheet.js]: I'm assuming it'll work if you check the node type ?

File: desktop/webpack.config.base.js (hunk 5 above)
Comment (chromakode) [inline desktop/webpack.config.base.js]: @chrisnojima IIRC Babel uses the location of `.babelrc` as the base to search for modules. Otherwise it looks relative to the file. So I get an error for both entry points like:

```
ERROR in ../shared/test/render-dumb-sheet.js
Module build failed: ReferenceError: Unknown plugin "transform-runtime" specified in "base" at 0, attempted to resolve relative to "/Users/mgood/Documents/dev/keybase/client/shared/test"
[...]

ERROR in ../shared/more/dumb-component-map.js
Module build failed: ReferenceError: Unknown plugin "transform-runtime" specified in "base" at 0, attempted to resolve relative to "/Users/mgood/Documents/dev/keybase/client/shared/more"
[...]
```

I'm not sure why this isn't happening for other webpack entry points, any ideas? Could also solve this by adding a `.babelrc` in the places this config is used (which might need to be duplicated in `shared`, which is why I went for this approach). `.babelrc` would be cleaner, though.

File: desktop/test/render-visdiff.js (hunk 4 above)
Comment (chromakode) [inline desktop/test/render-visdiff.js]: Nice cleanup! Committed here: https://github.com/keybase/client/pull/2788/commits/5e32460b999a30ad89b73e06237a722962de5ace

File: desktop/webpack.config.base.js (hunk 5 above)
Comment (chrisnojima) [inline desktop/webpack.config.base.js]: if you set ```nodePathDesktop: true,``` in your script in npm-helpers it should work. that adds the right node_path before running the script

Comment (chromakode) [inline desktop/webpack.config.base.js]: Oh, wacky. Yep, that works. Pull req updated.

**Comment by chrisnojima** [comment]
👍

File: shared/test/render-dumb-sheet.js (hunk 2 above)
Comment (chromakode) [inline shared/test/render-dumb-sheet.js]: `Node` doesn't have `getBoundingClientRect`, only `HTMLElement` does.
